### PR TITLE
Superficial cleanup

### PR DESCRIPTION
**Pull request by iskandr**
In preparation for more substantial changes in the next few days, I did a PEP8/style/readability pass over Vaxrank and made changes shouldn't change functionality. 

* Removed unused imports
* Fixed whitespace/indentation/line length
* Updated copyright years
* Moved FM index code into `reference_proteome` module, to hide implementation details from rest of logic (the calling code should only care that we're checking whether a peptide is in the reference, not we're using an FM index which returns all search results as a list).
 
The biggest change is in the `GenePathwayCheck` class, where I renamed things like `_CLASS_I_MHC` to `_CLASS_I_MHC_COLUMN_NAME` (made the code easier for me to read) and have the objects hold on to sets of the gene names and variants, rather than doing more complicated DataFrame operations for each variant.

### CODE REVIEW

File: vaxrank/gene_pathway_check.py [hunk 1]
```diff
@@ -54,72 +58,74 @@ def __init__(
...
         ----------
         variant : varcode.Variant
             Variant object to evaluate
         """
-        variant_dict = OrderedDict([
-            (_IFG_RESPONSE, False),
-            (_CLASS_I_MHC, False),
-            (_DRIVER_GENE, False),
-            (_DRIVER_VARIANT, False),
+        effect_description = variant.effects().top_priority_effect().short_description
```
[the remark this answers is not in the document]
Comment (iskandr) [reply vaxrank/gene_pathway_check.py]:
```
contig,start,ref,alt,is_coding_nonsynonymous,rna_support,mhc_binder,gene_name,interferon_gamma_response,class1_mhc_presentation_pathway,cancer_driver_gene,cancer_driver_variant
11,101177505,T,C,False,False,False,Cntnap1,False,False,False,False
13,5864876,,G,True,False,False,Klf6,False,False,False,False
4,45802539,G,C,True,False,False,Aldh1b1,False,False,False,False
9,82927102,G,T,True,True,True,Phip,False,False,False,False
X,8125624,C,A,True,True,True,Wdr13,False,False,False,False
```

[the remark this answers is not in the document]
Comment (iskandr) [reply vaxrank/gene_pathway_check.py]: The last 4 columns are always False in these -- is that expected?

[the remark this answers is not in the document]
Comment (iskandr) [reply vaxrank/gene_pathway_check.py]: I guess, given that all of our gene IDs are for humans and not mice, then they should be False.

File: vaxrank/epitope_prediction.py [hunk 2]
```diff
@@ -84,15 +86,15 @@ def fm_index_path(genome):
...
 def index_contains_kmer(fm, kmer):
     """
     Checks FM index for kmer, returns true if found.
     """
-    found = False
     for _ in fm.search(kmer):
-        found = True
-        break
-    return found
+        return True
```
Comment (julia326) [inline vaxrank/epitope_prediction.py]: can just say `return fm.search(kmer)` and get rid of this function, it's only used once

Comment (iskandr) [reply vaxrank/epitope_prediction.py]: I think it returns an iterator over matches, not sure how the boolean conversion logic interacts with that.

**Pull request merged**